### PR TITLE
[master] KCAL-18: Allow branch_variable module to search allowed doc types

### PR DESCRIPTION
**Pull request by swysor**
Also support array format for the variable path in the api schema.

### CODE REVIEW

**Comment by swysor** [comment]
crossbar pr is https://github.com/2600hz/kazoo-crossbar/pull/95

File: src/callflow.hrl [hunk 1]
```diff
@@ -40,6 +40,16 @@
 
 -define(NO_MATCH_CF, <<"no_match">>).
 
+-define(ALLOWED_BRANCH_DOC_TYPES, kapps_config:get_ne_binaries(?CF_CONFIG_CAT
```
Comment (jamesaimonetti) [inline src/callflow.hrl:43]: doc types are plural here, but singular in cf_callflow?

File: src/module/cf_branch_variable.erl [hunk 2]
```diff
@@ -41,12 +41,17 @@
 %%------------------------------------------------------------------------------
 -spec handle(kz_json:object(), kapps_call:call()) -> 'ok'.
 handle(Data, Call) ->
-    Scope = kz_json:get_ne_binary_value(<<"scope">>, Data, <<"custom_channel_vars">>),
     Variable = normalize_variable(kz_json:get_ne_value(<<"variable">>, Data)),
-    lager:info("looking for variable '~p' value in scope '~s' to find next callflow branch", [Variable, Scope]),
-    ChildName = find_child_in_scope(Scope, Variable, Call),
+    Scope = kz_json:get_ne_binary_value(<<"scope">>, Data, <<"custom_channel_vars">>),
+    Id = kz_json:get_ne_binary_value(<<"id">>, Data),
```
Comment (jamesaimonetti) [inline src/module/cf_branch_variable.erl]: `kz_doc:id(Data)` please

File: src/callflow.hrl (hunk 1 above)
Comment (jamesaimonetti) [reply src/callflow.hrl:43]: I believe doc types are singular (`kzd_devices:type() = <<"device">>` for instance).

Comment (swysor) [reply src/callflow.hrl:43]: oh good catch thanks!

**Pull request merged**